rp2040-hal/src/xosc.rs

```diff
@@ -107,20 +100,19 @@ impl CrystalOscillator<Disabled> {
         });
 
         //1 ms = 10e-3 sec and Freq = 1/T where T is in seconds so 1ms converts to 1000Hz
-        let delay_to_hz: Hertz = STABLE_DELAY.to_rate().map_err(|_| Error::BadArgument)?;
+        let delay_to_hz: HertzU32 = STABLE_DELAY.try_into_rate().ok_or(Error::BadArgument)?;
```

Comment:
You could move the `try_into_rate` call into line 89 (where STABLE_DELAY is declared..

Also you can just unwrap since we know the result is always ok and even if it isn't. It was not a bad argument...

Reply:
That's true. I also realize we could also store `STABLE_DELAY` directly as a HertzU32 and skip all that.